landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+    ... )
+    >>> latero = LateralEroder(mg, latero_mech="UC", Kv=0.001, Kl_ratio=1.5)
+
+    Run one step of flow accumulation and lateral erosion to get the dzlat array
+    needed for the next part of the test.
+
+    >>> fa.run_one_step()
+    >>> mg, dzlat = latero.run_one_step(dt)
+
+    Evolve the landscape until the first occurence of lateral erosion. Save arrays
+    volume of lateral erosion and topographic elevation before and after the first
+    occurence of lateral erosion
+
+    >>> while min(dzlat) == 0.0:
+    ...     oldlatvol = mg.at_node["volume__lateral_erosion"].copy()
+    ...     oldelev = mg.at_node["topographic__elevation"].copy()
+    ...     fa.run_one_step()
+    ...     mg, dzlat = latero.run_one_step(dt)
+    ...     newlatvol = mg.at_node["volume__lateral_erosion"]
+    ...     newelev = mg.at_node["topographic__elevation"]
+    ...     mg.at_node["topographic__elevation"][mg.core_nodes] += U * dt
+
+    Before lateral erosion occurs, *volume__lateral_erosion* has values at
+    nodes 6 and 10.
+
+    >>> np.around(oldlatvol, decimals=0)
+    array([  0.,   0.,   0.,   0.,
+             0.,   0.,  79.,   0.,
+             0.,   0.,  24.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.])
+
+
+    After lateral erosion occurs at node 6, *volume__lateral_erosion* is reset to 0
+
+    >>> np.around(newlatvol, decimals=0)
+    array([  0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,  24.,   0.,
+             0.,   0.,   0.,   0.,
+             0.,   0.,   0.,   0.])
+
+
+    After lateral erosion at node 6, elevation at node 6 is reduced by -1.41
+    (the height stored in dzlat[6]).
+
+    >>> np.around(oldelev, decimals=2)
+    array([ 0.  ,  1.03,  2.03,  3.  ,
+            1.04,  1.77,  2.45,  4.08,
+            2.09,  2.65,  3.18,  5.09,
+            3.04,  3.65,  4.07,  6.03,
+            4.03,  5.09,  6.02,  7.1 ])
+
+    >>> np.around(newelev, decimals=2)
+    array([ 0.  ,  1.03,  2.03,  3.  ,
+            1.04,  1.77,  1.03,  4.08,
+            2.09,  2.65,  3.18,  5.09,
+            3.04,  3.65,  4.07,  6.03,
+            4.03,  5.09,  6.02,  7.1 ])
+
+    >>> np.around(dzlat, decimals=2)
+    array([ 0.  ,  0.  ,  0.  ,  0.  ,
+            0.  ,  0.  , -1.41,  0.  ,
+            0.  ,  0.  ,  0.  ,  0.  ,
+            0.  ,  0.  ,  0.  ,  0.  ,
+            0.  ,  0.  ,  0.  ,  0. ])
+    """
+
+    _name = "LateralEroder"
+
+    _input_var_names = (
+        "topographic__elevation",
+        "drainage_area",
+        "flow__receiver_node",
+        "flow__upstream_node_order",
+        "topographic__steepest_slope",
+    )
+
+    _output_var_names = ("topographic__elevation",
+         "dzlat",
+         "vollat",
+         "qs_in"
+     )
+    _var_units = {
+        "topographic__elevation": "m",
+        "drainage_area": "m2",
+        "flow__receiver_node": "-",
+        "flow__upstream_node_order": "-",
```

Comment:
Does the LateralEroder require a route to one flow routing method (e.g., D8/D4), or can it work with multiple flow direction routing. If the former, I'd recommend writing an if statement in the __init__ that checks. See, for example [this similar test in a stream power component](https://github.com/landlab/landlab/blob/master/landlab/components/stream_power/fastscape_stream_power.py#L203).

Reply:
Similarly if LaterEroder ONLY works with RasterModelGrid's then this should be stated and it should be asserted that the grid is a RasterModelGrid.

Reply:
LateralEroder works with D8 or D4, but only with RasterModelGrid. A new assertion statement was added and description in parameter definitions was edited to reflect this.